packages/core/src/agent/MessageSender.ts

```diff
@@ -408,13 +419,59 @@ export class MessageSender {
     const outboundPackage = await this.packMessage(agentContext, { message, keys, endpoint: service.serviceEndpoint })
     outboundPackage.endpoint = service.serviceEndpoint
     outboundPackage.connectionId = connection?.id
-    for (const transport of this.outboundTransports) {
-      const protocolScheme = getProtocolScheme(service.serviceEndpoint)
-      if (!protocolScheme) {
-        this.logger.warn('Service does not have valid protocolScheme.')
-      } else if (transport.supportedSchemes.includes(protocolScheme)) {
-        await transport.sendMessage(outboundPackage)
-        return
+
```

Comment:
There's some duplicate flows now, so I think we may have to clean the `sendMessage` method up a bit. 

In the sendMessage method we do the following:
- check if there's a session to be found and if the message has return routing, if so return the message over the session
- if not returned over the session, loop through the services and call `this.sendToService` for each of the services in the did document.
- if the message could not be delivered, and the did document has a queue endpoint, we queu the message


The in the `sendToService` method we do the following:
- Check if there's outbound transports, and if so try to use the outbound transports
- If there's no outbound transport check if there's a session and try to send it over the sessionn.


There's a few improvements to this flow I think we can make. Especially since the message sender is already  one of the most complex classes.
- If you follow the flow above, `sendMessage` now has logic for sending to a session, but then the `sendToService` message as well. This means we now have two flows for sending a message to a session on a single path (sendMessage calls sendToService as well)
- The way sendToService handles session vs transports is that it will prefer outbound transports if they're available and otherwise fall back to the session sending. If we look at the flow for sendMessage above, it's the opposite: we first try to send it to the session, and then we start using the transports. I think we should keep the flows as similar as possible, and reduce duplciated code/flows as much as possible.

My suggestion would be to:
- Remove the session related code you added to `sendToService`. This way the `sendToService` keeps focused on one task: sending a message to a service, not dealing with session and inbound transports.
- Create a new private method `findSessionForOutboundContext` that will receive the outbound context, it will look for a session (based on the sessionId, the inboundSessionId and the connection in that order, based on what's availalbe). Then if a session has been found and if the message has return routing,  we return the session. This means if this method returns something we're certain we can use it to send a response.
- Update the `sendMessage` method to use this new `findSessionForOutboundContext` method, and if it returns a session, send the message to a session (this is alraedy implemented in `sendMessageToSession` and can be called). This logic is mostly already present in the `sendMessage` method and we can extract that into a new method.
- Extend the publcic `sendMessageToService` method to also include the `findSessionForOutboundContext` and `sendMessageToSession`. If no session is avaialble, or sending to the session failed we call `sendToService`.

We may want to rename the public `sendMessageToService` at some point to indicate we also support sending to a session (or combine it with sendMessage as the input api is 99% the same), but for now I think the above changes are enough.